packages/protocol/contracts/common/MentoFeeHandlerSeller.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity ^0.5.13;
+
+import "../common/interfaces/IFeeHandlerSeller.sol";
+
+import "../stability/interfaces/IExchange.sol";
+import "../stability/interfaces/ISortedOracles.sol";
+import "./UsingRegistry.sol";
+import "../stability/StableToken.sol";
+import "../common/FixidityLib.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "../common/Initializable.sol";
+
+contract MentoFeeHandlerSeller is IFeeHandlerSeller, UsingRegistry, Initializable {
+  using FixidityLib for FixidityLib.Fraction;
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialisation.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  function initialize(address _registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(_registryAddress);
+  }
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return Storage version of the contract.
+   * @return Major version of the contract.
+   * @return Minor version of the contract.
+   * @return Patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  function sell(
+    address sellTokenAddress,
+    address buyToken,
+    uint256 amount,
+    uint256 maxSlippage // as fraction,
+  ) external {
+    require(
+      buyToken == registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID),
+      "Buy token can only be gold token"
+    );
+
```

Comment:
Since this is external function we could add a check whether the contract has enough of sellToken